### PR TITLE
Add piping support

### LINKED ISSUE #1: Option to pipe data to stdout
I suggest an option to pipe the image data to stdout.

### PR DESCRIPTION
**Pull request by iamnapo**
Fixed #1 

I happened to use it, saw the issue and thought to add it 😛

### CODE REVIEW

**Comment by iamnapo** [comment]
@sindresorhus what do you think?

**Comment by kevva** [comment]
Instead of an option, couldn't we just check if `process.stdout.isTTY` is `false`?

**Comment by iamnapo** [comment]
I didn't think of that, should I change it?

**Comment by kevva** [comment]
Yup. If you can avoid introducing a new option, it's good :).

**Comment by iamnapo** [comment]
Thanks for the feedback!
I'm not sure how to use execa to test for the default option (return icon's location) now.

File: readme.md [hunk 1]
```diff
@@ -2,40 +2,38 @@
...
-```
+```bash
 $ npm install --global file-icon-cli
 ```
 
-
 ## Usage
 
-```
+```bash
```
Comment (kevva) [inline readme.md]: Remove this and the one before.

File: readme.md [hunk 2]
```diff
@@ -2,40 +2,38 @@
...
     $ file-icon <app-name|bundle-id|file-path>
+    $ file-icon <app-name|bundle-id|file-path> > icon.png
 
   Options
     --size  Size of the icon [Max: 1024]
 
   Examples
     $ file-icon Safari
     /tmp/86ca9400-9f34-4a64-ab24-027d80f88b46/icon.png
+    $ file-icon Safari > icon.png
```
Comment (kevva) [inline readme.md]: Move this to the bottom instead. Looks better with it not having any logged output.

File: test.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import test from 'ava';
 import execa from 'execa';
 
-test(async t => {
-	t.true((await execa.stdout('./cli.js', ['Safari'])).endsWith('icon.png'));
+test(`outputs buffer when piped`, async t => {
+	t.true((await execa.stdout('./cli.js', ['Safari'])).startsWith(Buffer.from([0x89, 0x50, 0x4E, 0x47, 0x0D, 0x0A, 0x1A, 0x0A])));
```
Comment (kevva) [inline test.js]: I would use `file-icon` in here and get the icon buffer first and then compare it to the one outputted by the CLI.

**Review by kevva — CHANGES_REQUESTED** [review]

**Comment by kevva** [comment]
> I'm not sure how to use execa to test for the default option (return icon's location) now.

You could just set `process.stdout.isTTY` to `true` manually in the test and then reset it to whatever it was before.

**Comment by iamnapo** [comment]
> You could just set `process.stdout.isTTY` to `true` manually in the test and then reset it to whatever it was before.

Inside the test it's already true, the problem is inside the child process that is created.

**Comment by kevva** [comment]
Oh right, you could pass `stdio: 'inherit'` to `execa` and it should work.

**Comment by iamnapo** [comment]
It does, but I can't capture the output 😕

**Comment by iamnapo** [comment]
Then `process.stdout.isTTY` is `undefined`. No? 😕

**Comment by iamnapo** [comment]
I think we can skip this test after all.

**Comment by iamnapo** [comment]
Friendly ping @kevva :relaxed:

**Comment by iamnapo** [comment]
Friendly ping @kevva @sindresorhus :relaxed: (I think we can skip this test, after all)

**Comment by sindresorhus** [comment]
The trick to make it work in `execa` is to not check `process.stdout.isTTY`, but rather to just assume piping when there are no arguments.

File: test.js [hunk 4]
```diff
@@ -1,6 +1,8 @@
 import test from 'ava';
 import execa from 'execa';
+import fileIcon from 'file-icon';
 
-test(async t => {
-	t.true((await execa.stdout('./cli.js', ['Safari'])).endsWith('icon.png'));
```
Comment (sindresorhus) [inline test.js:5]: Please preserve the existing test.

File: readme.md [hunk 5]
```diff
@@ -28,14 +27,13 @@ $ file-icon --help
...
     /tmp/c3871faa-d759-48b9-ac85-5504d712a02a/icon.png
+    $ file-icon Safari > icon.png
 ```
 
-
 ## Related
 
 - [file-icon](https://github.com/sindresorhus/file-icon) - API for this module
 
-
```
Comment (sindresorhus) [inline readme.md]: Don't do unrelated changes.

File: package.json [hunk 6]
```diff
@@ -40,13 +40,13 @@
 		"swift"
 	],
 	"dependencies": {
-		"file-icon": "^2.0.1",
-		"meow": "^3.7.0",
-		"tempy": "^0.2.0"
+		"file-icon": "^2.1.0",
+		"meow": "^4.0.0",
+		"tempy": "^0.2.1"
```
Comment (sindresorhus) [inline package.json]: Don't do unrelated changes.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by iamnapo** [comment]
but even when piping, there will be arguments. No?

**Comment by sindresorhus** [comment]
Ah yeah, that's true... I guess we can just pass down an environment variable in the test and check for it in the code. Just for testing. Like `__FILE_ICON_SHOULD_PIPE__`. It's ugly, but at least then we can test it.

**Comment by iamnapo** [comment]
Hmm. The problem is that `execa` runs the command in a child process, and in that child process `process.stdout.isTTY` is always `false`. So maybe you meant an env variable `__FILE_ICON_SHOULD_NOT_PIPE__`? If so, I think that an option like `--no-pipe` would be better: we can test it and lets someone pipe the location of the image as well.

**Comment by sindresorhus** [comment]
> So maybe you meant an env variable __FILE_ICON_SHOULD_NOT_PIPE__? 

Yes, I said environment variable.

> If so, I think that an option like --no-pipe would be better: we can test it and lets someone pipe the location of the image as well.

I don't want to expose it as a flag to users. We only need this workaround for testing.

**Comment by sindresorhus** [comment]
Thanks for your patience, @iamnapo. It looks good now :)

**Pull request merged**